### ISSUE TITLE
In the settle page remove exchanges of values: 0

### ISSUE THREAD

**Issue by Natim** [spiral-project/ihatemoney#138]

**Comment by zorun** [comment]
I saw this happen too when transferring money to settle things manually (see #139 ).  The "Settle" page then has some lines with an amount equal to 0.00.

I think it may be related to rounding issues (i.e. the amount is displayed as "0.00" but is actually equal to e.g. 0.000001).

**Comment by zorun** [comment]
This causes the issue: https://github.com/spiral-project/ihatemoney/blob/master/budget/models.py#L219

The amount of a bill should be a Decimal object instead of a float, see https://docs.python.org/2/library/decimal.html

SQLAlchemy seems to support Decimal objects, however it would require a database migration: http://docs.sqlalchemy.org/en/latest/core/type_basics.html#sqlalchemy.types.Numeric

**Comment by Natim** [comment]
We can also round it for the comparaison to avoid database migration

**Comment by zorun** [comment]
@natim yes, that would work for this issue, but similar hackish solutions would be needed elsewhere.

I tried to switch to a Numeric datatype, but it does not seem to solve the issue.  This warning may explain why:

```
/tmp/venv2-ihatemoney/lib/python2.7/site-packages/sqlalchemy/sql/sqltypes.py:596: SAWarning: Dialect sqlite+pysqlite does *not* support Decimal objects natively, and SQLAlchemy must convert from floating point - rounding errors and other issues may occur. Please consider storing Decimal numbers as strings or integers on this platform for lossless storage.
```

Basically, when using sqlite3, it's still using floats inside the database, so rounding errors still occur.

Another solution could be to switch to integers (as suggested by the warning), for instance storing a number of cents as integer.

**Comment by aavenel** [comment]
This is not the first time we have some issues about rounding.  See #126 

I agree with @zorun, if we can't use Decimal, the proper fix would be to store the amount as an integer. However, this requires some database migration. Perhaps it's not worth it and we can add a round() like @Natim suggested.

**Comment by zorun** [comment]
@aavenel Yes, it's a large change, and it would bring other issues.  I have moved the general discussion to a new ticket #165 

The initial issue should be fixed by #164
